addons/mail/static/src/discuss/core/common/notification_settings.js

```diff
@@ -0,0 +1,98 @@
+/* @odoo-module */
+
+import { useState, Component } from "@odoo/owl";
+import { Dropdown } from "@web/core/dropdown/dropdown";
+import { DropdownItem } from "@web/core/dropdown/dropdown_item";
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { useService } from "@web/core/utils/hooks";
+
+const notificationTypeRegistry = registry.category("mail.discuss/notification_types");
+notificationTypeRegistry
+    .add("all", {
+        name: _t("All Messages"),
+    })
+    .add("mentions", {
+        name: _t("Mentions Only"),
+    })
+    .add("none", {
+        name: _t("Nothing"),
+    });
+const muteDurationRegistry = registry.category("mail.discuss/mute_durations");
+muteDurationRegistry
+    .add("15_mins", {
+        value: 15,
+        name: _t("For 15 minutes"),
+    })
+    .add("1_hour", {
+        value: 60,
+        name: _t("For 1 hour"),
+    })
+    .add("3_hours", {
+        value: 180,
+        name: _t("For 3 hours"),
+    })
+    .add("8_hours", {
+        value: 480,
+        name: _t("For 8 hours"),
+    })
+    .add("24_hours", {
+        value: 1440,
+        name: _t("For 24 hours"),
+    })
+    .add("forever", {
+        value: -1,
+        name: _t("Until I turn it back on"),
+    });
+
```

Comment:
I don't see any value for using registries: they are used only once, and things are defined only in this file.

Reply:
The idea was to use in here and also in the following PR ( general settings for `Discuss` ).

Reply:
I would not introduce registry, because the API sucks and using formatters on them makes the code way harder to read than it should.
If this makes sense to add in following PR, let's do in following PR. I don't see any value here, and if following PR still adds not value of registry, it's gonna be harder to motivate simplify the code by making more diffs in these follow-up PRs